R/metadata.R

```diff
@@ -95,30 +128,28 @@ imeta <- function(
   invisible(resp)
 }
 
-# measure depth of list (https://stackoverflow.com/questions/13432863/determine-level-of-nesting-in-r)
-list_depth <- function(this, thisdepth = 0) {
-  if(!is.list(this)) {
-    thisdepth
-  } else {
-    max(unlist(lapply(this, list_depth, thisdepth = thisdepth + 1)))
-  }
-}
-
-# TODO add some reference to documentation to how to query?
 #' Query data objects and collections in iRODS
 #'
-#' Use SQL-like expressions to query data objects and collections based on different properties.
+#' Use SQL-like expressions to query data objects and collections based on
+#' different properties. Read more about queries by looking at the iCommands
+#' equivalent `iquest` in the [iRODS Docs](https://docs.irods.org/master/icommands/user/#iquest).
 #'
 #' @param query GeneralQuery for searching the iCAT database.
 #' @param limit Maximum number of rows to return. Defaults to 100.
 #' @param offset Number of rows to skip for paging. Defaults to 0.
 #' @param type Type of query: 'general' (the default) or 'specific'.
-#' @param case_sensitive Whether the string matching in the query is case sensitive.
-#'   Defaults to `TRUE`.
-#' @param distinct Whether only distinct rows should be listed. Defaults to `TRUE`.
-#' @param verbose Whether information should be printed about the HTTP request and response.
+#' @param case_sensitive Whether the string matching in the query is case
+#'  sensitive. Defaults to `TRUE`.
+#' @param distinct Whether only distinct rows should be listed. Defaults to
+#'  `TRUE`.
+#' @param verbose Whether information should be printed about the HTTP request
```

Comment:
defaults to false

Reply:
On the R side this defaults now to TRUE. Should it be set to FALSE to match icommand iquest?

Reply:
oh, i don't know, was more pointing out that the others named their default value in the documentation.   if TRUE is the default today, i think we should at least say that here.

whether is should be FALSE... not sure.

Reply:
changed to include default value. I noted that default values were omitted at more places, so I included those as well. I guess it is best to leave this as an additional commit.

Reply:
Oh, ok, thanks! I normally don't specify the default value in the @param description because it's already in the function call (which is also visible in the doc). But of course it can stay if it makes it more clear.

Reply:
I guess duplication is not bad. Also, not all users will look at the function call or understand it in that way.

Reply:
Sure!